textual_fspicker/parts/drive_navigation.py

```diff
@@ -0,0 +1,116 @@
+class DriveEntry(Option):
+    """A drive entry for the `DriveNavigation` class."""
+
+    def __init__(self, drive: Path | str) -> None:
+        self.drive_root: Path = MakePath.of(drive)
+        """The drive root for this entry."""
+        super().__init__(self.drive_root.drive, id=self.drive_root.drive)
+
+
+class DriveNavigation(OptionList):
+    """A drive navigation widget.
+
+    Provides a single-pane widget that lets the user select a drive. This is
+    very useful in combination with the `DirectoryNavigation` widget. A dialog can
+    reload that widget in response to drive selection changes.
+    """
+
+    DEFAULT_CSS = """
+    DriveNavigation, DriveNavigation:focus {
+        border: blank;
+        border-right: $panel-darken-1;
+        width: 10;
+        height: 100%;
+    }
+
+    DriveNavigation > .option-list--option-highlighted {
+        text-style: bold;
+        color: white;
+    }
+    """
+    """Default styling for the widget."""
+
+    @dataclass
+    class DriveSelected(Message):
+        """Message sent when a drive is selected."""
+
+        drive_root: Path
+        """The selected drive root, like `c:\\`."""
+
+    drive: var[str] = var[str](MakePath.of(".").absolute().drive, init=False)
+
+    def __init__(self, location: Path | str = ".") -> None:
+        """Initialise the drive navigation widget.
+
+        Args:
+            location: The starting location.
+        """
+        super().__init__()
+        self.set_reactive(DriveNavigation.drive, MakePath.of(location).absolute().drive)
+        if platform.system() == "Windows":
+            self._entries = [DriveEntry(drive) for drive in os.listdrives()]
+
+    def on_mount(self) -> None:
+        """Add available drives to the widget."""
+        self.add_options(self._entries)
+        self.highlight_drive(self.drive)
+
+    def _watch_drive(self, drive: str) -> None:
+        """Highlight the new drive.
+
+        Args:
+            drive: The new value of the current drive.
+        """
+        self.highlight_drive(drive)
+
+    def highlight_drive(self, drive: str) -> None:
+        """Highlight the given drive.
+
+        Args:
+            drive: The drive to be highlighted.
+
+        Raises:
+            RuntimeError: Raised if there is no entry for the drive.
+        """
+        for entry in self._entries:
+            if entry.drive_root.drive == drive:
+                break
+        else:
+            raise RuntimeError(f"Drive {drive} not found in DriveNavigation widget.")
+        self.highlighted = self.get_option_index(entry.id)
+
+    @on(OptionList.OptionSelected)
+    def drive_selected(self, event: OptionList.OptionSelected) -> None:
+        """Post a DriveSelected message.
+
+        Args:
+            event: The drive selected event from the parent `OptionList`.
+        """
+        assert isinstance(event.option, DriveEntry)
+        self.post_message(self.DriveSelected(drive_root=event.option.drive_root))
```

Comment:
It would be a good idea to put:

```python
event.stop()
```

before posting the message here, so that the `OptionList.OptionSelected` doesn't leak up the DOM.